pkg/report/table/summary.go

```diff
@@ -0,0 +1,308 @@
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/scanner/langpkg"
+	"github.com/aquasecurity/trivy/pkg/types"
+	xstrings "github.com/aquasecurity/trivy/pkg/x/strings"
+)
+
+type Scanner interface {
+	Header() string
+	Alignment() table.Alignment
+
+	// Count returns the number of findings, but -1 if the scanner is not applicable
+	Count(result types.Result) int
+
+	String() string // Required to show correct logs
+}
+
+func NewScanner(scanner types.Scanner) Scanner {
+	switch scanner {
+	case types.VulnerabilityScanner:
+		return VulnerabilityScanner{}
+	case types.MisconfigScanner:
+		return MisconfigScanner{}
+	case types.SecretScanner:
+		return SecretScanner{}
+	case types.LicenseScanner:
+		return LicenseScanner{}
+	}
+	return nil
+}
+
+type scannerAlignment struct{}
+
+func (s scannerAlignment) Alignment() table.Alignment {
+	return table.AlignCenter
+}
+
+type VulnerabilityScanner struct{ scannerAlignment }
+
+func (s VulnerabilityScanner) Header() string {
+	return "Vulnerabilities"
+}
+
+func (s VulnerabilityScanner) Count(result types.Result) int {
+	if result.Class == types.ClassOSPkg || result.Class == types.ClassLangPkg {
+		return len(result.Vulnerabilities)
+	}
+	return -1
+}
+
+func (s VulnerabilityScanner) String() string {
+	return string(types.VulnerabilityScanner)
+}
+
+type MisconfigScanner struct{ scannerAlignment }
+
+func (s MisconfigScanner) Header() string {
+	return "Misconfigurations"
+}
+
+func (s MisconfigScanner) Count(result types.Result) int {
+	if result.Class == types.ClassConfig {
+		return len(result.Misconfigurations)
+	}
+	return -1
+}
+
+func (s MisconfigScanner) String() string {
+	return string(types.MisconfigScanner)
+}
+
+type SecretScanner struct{ scannerAlignment }
+
+func (s SecretScanner) Header() string {
+	return "Secrets"
+}
+
+func (s SecretScanner) Count(result types.Result) int {
+	if result.Class == types.ClassSecret {
+		return len(result.Secrets)
+	}
+	return -1
+}
+
+func (s SecretScanner) String() string {
+	return string(types.SecretScanner)
+}
+
+type LicenseScanner struct{ scannerAlignment }
+
+func (s LicenseScanner) Header() string {
+	return "Licenses"
+}
+
+func (s LicenseScanner) Count(result types.Result) int {
+	if result.Class == types.ClassLicense || result.Class == types.ClassLicenseFile {
+		return len(result.Licenses)
+	}
+	return -1
+}
+
+func (s LicenseScanner) String() string {
+	return string(types.LicenseScanner)
+}
+
+type summaryRenderer struct {
+	w          *bytes.Buffer
+	isTerminal bool
+	scanners   []Scanner
+}
+
+func NewSummaryRenderer(buf *bytes.Buffer, isTerminal bool, scanners types.Scanners) *summaryRenderer {
+	if !isTerminal {
+		tml.DisableFormatting()
+	}
+
+	var ss []Scanner
+	for _, scanner := range scanners {
+		s := NewScanner(scanner)
+		if lo.IsNil(s) {
+			continue
+		}
+		ss = append(ss, s)
+	}
+
+	return &summaryRenderer{
+		w:          buf,
+		isTerminal: isTerminal,
+		scanners:   ss,
```

Comment:
```suggestion
		scanners:   ss,
		logger: log.WithPrefix("report"),
```

Reply:
changed in https://github.com/aquasecurity/trivy/pull/8177/commits/67a729cd2b36e3171316c26133da2e5a15c8b9a2